slave.mk

```diff
@@ -264,6 +266,44 @@ ifeq ($(CONFIGURED_PLATFORM),vs)
 export BUILD_MULTIASIC_KVM
 endif
 
+ifeq ($(CROSS_BUILD_ENVIRON),y)
+DEB_BUILD_OPTIONS_GENERIC += nocheck
+export $(dpkg-architecture -a$(CONFIGURED_ARCH))
+ifeq ($(ENABLE_PY2_MODULES),n)
+ANT_DEB_CROSS_PROFILES=nopython2
+endif
+ANT_DEB_CROSS_OPT := -a$(CONFIGURED_ARCH) -Pcross,nocheck,$(ANT_DEB_CROSS_PROFILES)
+ANT_DEB_CONFIG := CONFIG_SITE=/etc/dpkg-cross/cross-config.$(CONFIGURED_ARCH)
+
+VIRTENV_BASE_CROSS_PYTHON2 = /python_virtualenv/env2/
+VIRTENV_BASE_CROSS_PYTHON3 = /python_virtualenv/env3/
+VIRTENV_BIN_CROSS_PYTHON2 = $(VIRTENV_BASE_CROSS_PYTHON2)/bin/
+VIRTENV_BIN_CROSS_PYTHON3 = $(VIRTENV_BASE_CROSS_PYTHON3)/bin/
+VIRTENV_LIB_CROSS_PYTHON2 = $(VIRTENV_BASE_CROSS_PYTHON2)/lib/
+VIRTENV_LIB_CROSS_PYTHON3 = $(VIRTENV_BASE_CROSS_PYTHON3)/lib/
+
+ifeq ($(CONFIGURED_ARCH),armhf)
+CROSS_HOST_TYPE = arm-linux-gnueabihf
+GOARCH=arm
+else ifeq ($(CONFIGURED_ARCH),arm64)
+CROSS_HOST_TYPE = aarch64-linux-gnu
+GOARCH=arm64
+endif
+
+CROSS_COMPILE = $(CROSS_HOST_TYPE)-
+CROSS_CC = $(CROSS_COMPILE)gcc
+CROSS_CXX = $(CROSS_COMPILE)g++
+CROSS_AR = $(CROSS_COMPILE)ar
```

Comment:
Instead of using `CROSS_CC`/`CROSS_CXX`/`CROSS_AR`, can `CC`/`CXX`/`AR` be used directly, and exported as environment variables? That way, multiple places later on won't need lines like `CC=$(CROSS_CC)`.